scripts/with_pygithub/notifier.py

```diff
@@ -0,0 +1,26 @@
+import argparse
+
+
+from github import Github
+
+DOC_LABEL = "To be documented"
+# DOC_MSG = "@intellij-rust/technical-writers, seems like it's time to write the documentation!"
+DOC_MSG = "Docteam, seems like it's time to write the documentation!"
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--token", type=str, required=True)
+    parser.add_argument("--pull_request", type=int, required=True)
+    parser.add_argument("--event", type=str, required=True)
+    parser.add_argument("--label", type=str)
+    parser.add_argument("--repo", type=str)
+
+    args = parser.parse_args()
+
+    g = Github(args.token)
+    repo = g.get_repo(args.repo)
+    pr = repo.get_pull(args.pull_request)
+
+    if (args.event == "closed" and pr.is_merged() and DOC_LABEL in [i.name for i in pr.labels]) \
```

Comment:
Instead of checking for `pr.is_merged()` in the script you can use `if: github.event.pull_request.merged` in the workflow file ([example](https://github.com/intellij-rust/intellij-rust/blob/06c55e069eec2ea29da3fbb0e9193c5d0257b764/.github/workflows/milestones.yml#L19)) - this way the script will only be called if the PR was merged.

Reply:
I suspect the whole `if` might be replaced with `if any(label.name == DOC_LABEL for label in pr.labels):` assuming you are checking for a merged PR in `yml`. Isn't it?

Reply:
>I suspect the whole if might be replaced with if any(label.name == DOC_LABEL for label in pr.labels): assuming you are checking for a merged PR in yml. Isn't it?

not really, the current workflow is triggered every time **any** label is added.
For example, if the user adds `to be documented` and `major` (means "QA, look here") in merged PR, then both times such a check will work and two comments will be written, not critical, but could be better :)

This logic can be added to `*.yml`, but I'm not sure if this will make the summary code more understandable.

Reply:
Ah, I see the culprit now, thanks.